languagetool-core/src/main/java/org/languagetool/JLanguageTool.java

```diff
@@ -1145,7 +1157,8 @@ protected void printSentenceInfo(AnalyzedSentence analyzedSentence) {
 
   protected List<RuleMatch> performCheck(List<AnalyzedSentence> analyzedSentences, List<String> sentences,
                                          List<Rule> allRules, ParagraphHandling paraMode, AnnotatedText annotatedText, Mode mode, Level level) throws IOException {
-    return performCheck(analyzedSentences, sentences, allRules, paraMode, annotatedText, null, mode, level, true);
+    List<Rule> nonIgnored = allRules.stream().filter(r -> !ignoreRule(r)).collect(Collectors.toList());
```

Comment:
This method is unused in LT, so I can remove it instead of changing. Not sure if there are external callers though.

Reply:
As LT can be used from Java directly via its API, we should expect public methods to be used. We can clean them up, but for that, we should first deprecate them and only remove them 2 releases later or so.

Reply:
This one is deprecated.
So, should I deprecate these methods?

Reply:
> So, should I deprecate these methods?

Yes